src/Analysis/Engine/Impl/Values/ClassInfo.cs

```diff
@@ -318,7 +318,7 @@ internal class ClassInfo : AnalysisValue, IClassInfo, IHasRichDescription, IHasQ
             );
         }
 
-        public IEnumerable<IAnalysisSet> Bases => _bases;
+        public IEnumerable<IAnalysisSet> Bases => _bases.AsLockedEnumerable();
```

Comment:
`LockedEnumerableObject` only takes the lock for enumerator. Pretty much it does nothing. What you need here is probably change the type of the `_bases` to array (or `ImmutableArray`) and inside `SetBases` simply replace the field value. `SetBase` can be removed, it isn't used anywhere.